parallel-install/pkg/preinstaller/resourcemanager.go

```diff
@@ -121,14 +129,48 @@ func (c *DefaultResourceManager) UpdateResource(resource *unstructured.Unstructu
 	return obj, nil
 }
 
-func (c *DefaultResourceManager) getResource(resourceName string, resourceSchema schema.GroupVersionResource) (*unstructured.Unstructured, error) {
-	return c.dynamicClient.Resource(resourceSchema).Get(context.TODO(), resourceName, metav1.GetOptions{})
+func (c *DefaultResourceManager) DeleteCollectionOfResources(gvk schema.GroupVersionKind, opts metav1.DeleteOptions, listOpts metav1.ListOptions) error {
+	var err error
+	err = retry.Do(func() error {
+		if err = c.deleteCollectionOfResources(retrieveGvrFrom(gvk), opts, listOpts); err != nil {
+			c.log.Errorf("Error occurred during resources delete: %s", err.Error())
+			return err
+		}
+
+		return nil
+	}, c.retryOptions...)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *DefaultResourceManager) getResource(resourceName string, gvr schema.GroupVersionResource, opts metav1.GetOptions) (*unstructured.Unstructured, error) {
+	return c.dynamicClient.Resource(gvr).Get(context.TODO(), resourceName, opts)
+}
+
+func (c *DefaultResourceManager) createResource(resource *unstructured.Unstructured, gvr schema.GroupVersionResource, opts metav1.CreateOptions) (*unstructured.Unstructured, error) {
+	return c.dynamicClient.Resource(gvr).Create(context.TODO(), resource, opts)
+}
+
+func (c *DefaultResourceManager) updateResource(resource *unstructured.Unstructured, gvr schema.GroupVersionResource, opts metav1.UpdateOptions) (*unstructured.Unstructured, error) {
+	return c.dynamicClient.Resource(gvr).Update(context.TODO(), resource, opts)
 }
 
-func (c *DefaultResourceManager) createResource(resource *unstructured.Unstructured, resourceSchema schema.GroupVersionResource) (*unstructured.Unstructured, error) {
-	return c.dynamicClient.Resource(resourceSchema).Update(context.TODO(), resource, metav1.UpdateOptions{})
+func (c *DefaultResourceManager) deleteCollectionOfResources(gvr schema.GroupVersionResource, opts metav1.DeleteOptions, listOpts metav1.ListOptions) error {
```

Comment:
What is the value of having all these unexported functions?

Wouldn't it be the same calling (e.g. in this case) `c.dynamicClient.Resource(gvr).DeleteCollection(...)` directly in the public function?

I fail to see the benefit of having this.

Reply:
In first iteration of `PreInstaller` I introduced these in order to somehow spy on the tested object and mock the function calls (not to call the `dynamicClient` itself at all. Later I discovered the way we can test it in go and use `fake` objects for testing purposes, but these functions were left the same. In fact I can put them to the exported one and we will end up with the same functionality. 👍